### PR TITLE
De-risking Guide v2: Add remaining resource pages and workshop call-to-action

### PR DESCRIPTION
**Pull request by nateborr**
## Changes proposed in this pull request:

Note that these changes will be added to the `derisking-government-tech` feature branch, and not yet merged into main or deployed.

### New content

Implement the remaining three resource appendix sections for the v2 de-risking guide.

- Sample kick-off week agenda
- Open source security
- Quality indicators plan (sample QASP)

With these changes, there is no more placeholder content in the v2 de-risking guide. There should be no instances of "TK" or "TODO" in the guide content.

### Workshop call-to-action

At the bottom of the primary v2 de-risking guide pages within the Home section, render a call-to-action offering workshop services with a "Contact us" button linking to the 18F contact page.

### Fix sub-nav issues on v2 de-risking guide resource pages

Update the "Market research" and "Verbal interview questions" resource pages to use `h2` elements for their subheadings rather than `h3`s. This corrects those pages' sub-nav menus to display the link for the current subheading in bold, and improves the semantic structure of the pages.

Note that on some pages, scrolling to the bottom does not highlight the last link in the sub-nav; I believe this is due to an existing bug in which the page's scroll position may not be able to reach as far as the last heading's offset from the top of the page. We shouldn't scope that issue in for the v2 de-risking guide release.

### Page layout

The workshop call-to-action is implemented as an HTML `section` outside of the main page content. It is optionally rendered within the `page` layout, if these two conditions are met:

1. the page's tags identify it as part of the v2 de-risking guide
2. the page's 11ty front matter sets the `show-workshop-cta` indicator variable to true

This should not effect the layout of pages in any of the other guides, but because the page layout file has changed they should be smoke-tested.

### PDF download link

This PR extracts a generic component from the download-as-PDF component that was added with the bid evaluation pages, so that it can be reused for similar functionality on the sample kick-off agenda page. It accepts the document descriptor (i.e. "worksheet" or "schedule") and the PDF asset link as parameters.

## Testing

Beyond the automated validations on the PR, manually test the following on a PR build:

### Newly implemented sections

For each of the `Kick-off agenda`, `Open source security`, and `Quality indicators plan` pages of the Resources section:

- Visit the /derisking-government-tech/resources/ route on the PR build URL and confirm you can reach the pages in question via the second row of cards on the page: https://federalist-8bc0b42c-5fd0-4ae8-9366-cd9c265a4446.sites.pages.cloud.gov/preview/18f/guides/drg-v2-add-remaining-resources/derisking-government-tech/resources/
- Confirm that the content, layout, and images on the page look correct
- Confirm that links on the page go to the expected destinations
- On the kick-off agenda page, confirm that the download link opens the PDF version of the sample schedule
- Check mobile device rendering using browser dev tools
- Shake out the page's responsiveness when the browser window is resized
- Confirm meta title, meta description, and social meta tags are present in the page source
- Shake out the top nav and, where present, the left nav

### Call-to-action

- Confirm that the "Contracting problems? We can help." CTA appears correctly at the bottom of the following pages, directly above the footer:
  - Introduction
  - Software solutions
  - Key principles
  - Buying custom development services
  - Vendor management
  - Conclusion
- Confirm that the "Contact us" button opens https://18f.gsa.gov/contact/
- Test the CTA's responsiveness when the viewport is resized and for mobile device sized views. The light bulb image should be removed on screens narrower than 640 pixels (the USWDS `tablet` breakpoint size).

### Resource pages with fixed sub-navs

For each of the "Market research" and "Verbal interview questions" pages:

- Confirm that scrolling down the page or selecting a sub-nav link sets the name of the current section in the sub-nav to bold. (Although note that this logic may not work for the final sub-nav item, as described above.)
- Review whether the heading styles are acceptable.

### Pages potentially affected by layout updates

- Spot-check a few pages in some of the other guides (for example, the v1 de-risking guide, the engineering guide, etc.) and confirm that the page footers still look as expected, *without* the new "Contracting problems? We can help." call-to-action. Some of these can be reached from the [This is not the Guides home page](https://federalist-8bc0b42c-5fd0-4ae8-9366-cd9c265a4446.sites.pages.cloud.gov/preview/18f/guides/drg-v2-add-remaining-resources/) base page in the PR build.
- Within the v2 de-risking guide, check that the PDF download links on the [/evaluate-bids/](https://federalist-8bc0b42c-5fd0-4ae8-9366-cd9c265a4446.sites.pages.cloud.gov/preview/18f/guides/drg-v2-add-remaining-resources/derisking-government-tech/resources/evaluate-bids/#evaluator-worksheet) and [/evaluator-worksheet/](https://federalist-8bc0b42c-5fd0-4ae8-9366-cd9c265a4446.sites.pages.cloud.gov/preview/18f/guides/drg-v2-add-remaining-resources/derisking-government-tech/resources/evaluator-worksheet/) pages still open the evaluator worksheet PDF

## security considerations

No security considerations: These updates use the existing framework and patterns of the 18F Guides site.

### CODE REVIEW

File: content/derisking-government-tech/resources/kickoff.md [hunk 1]
```diff
@@ -1,9 +1,180 @@
...
+  href: '#day-5'
 ---
 
-TK
+{% include 'derisking-government-tech/pdf_document_download.html' doc_type_name: "agenda" pdf_url: "/assets/derisking-government-tech/dist/18F-de-risking-guide-v2-sample-kick-off-week-agenda.pdf" %}
+
+---
+
+<table class="kick-off-table">
+  <caption><h3 id="day-1">Day 1</h3></caption>
```
Comment (melchoyce) [inline content/derisking-government-tech/resources/kickoff.md:32]: It looks like this page skips from an `h1` to `h3`s

File: content/derisking-government-tech/resources/kickoff.md [hunk 2]
```diff
@@ -1,9 +1,180 @@
...
+- text: Day 1
+  href: '#day-1'
+- text: Day 2
+  href: '#day-2'
+- text: Day 3
+  href: '#day-3'
+- text: Day 4
+  href: '#day-4'
+- text: Day 5
+  href: '#day-5'
```
Comment (melchoyce) [inline content/derisking-government-tech/resources/kickoff.md:24]: For some reason these nav items aren't bolding when you're on that section:

image

File: content/derisking-government-tech/resources/open-source-security.md [hunk 3]
```diff
@@ -3,7 +3,77 @@ title: Best practices for open source software security
 permalink: /derisking-government-tech/resources/open-source-security/
 layout: layouts/page
 tags: derisking-government-tech
-sidenav: false
+description: High-level explanation for keeping data, static assets, secrets, and code safe in an open source project.
+eleventyNavigation:
+  parent: derisking-government-tech-resources-open-source-security
+  key: open-source-security
+  order: 1
+  title: Best practices for open source software security
```
Comment (melchoyce) [inline content/derisking-government-tech/resources/open-source-security.md:11]: I think we should maybe shorten this to "Open source security" since it ends up wrapping onto three lines:

image

Comment (melchoyce) [inline content/derisking-government-tech/resources/open-source-security.md:11]:
```suggestion
  title: Open source security
```

**Review by melchoyce — CHANGES_REQUESTED** [review]

File: content/derisking-government-tech/resources/open-source-security.md (hunk 3 above)
Comment (melchoyce) [reply content/derisking-government-tech/resources/open-source-security.md:11]: Huh it actually looks like in the Resources section this behavior is _only_ working on the open source security page

File: content/derisking-government-tech/resources/kickoff.md [hunk 4]
```diff
@@ -1,9 +1,180 @@
...
 permalink: /derisking-government-tech/resources/kickoff/
 layout: layouts/page
 tags: derisking-government-tech
-sidenav: false
+description: Sample kick-off week agenda for starting work with a vendor development team building custom software for a government agency.
+eleventyNavigation:
+  parent: derisking-government-tech-resources-kickoff
+  key: kickoff
+  order: 1
+  title: Sample kick-off week agenda
```
Comment (melchoyce) [inline content/derisking-government-tech/resources/kickoff.md:11]: We could probably shorten this one too

```suggestion
  title: Kick-off agenda
```

File: content/derisking-government-tech/resources/kickoff.md (hunk 2 above)
Comment (nateborr) [reply content/derisking-government-tech/resources/kickoff.md:24]: I'll look into it. I manually added the anchors for those Day X headings so that I could put them into the table captions, so I may have made an error there.

Comment (nateborr) [reply content/derisking-government-tech/resources/kickoff.md:24]: The first issue is that I used `h3`s instead of `h2`s, as you pointed out.

When I fix that, there's another issue: The `setCurrentLink()` JS function that updates the current sub-nav selection works by comparing the current scroll position to the `offsetTop` value for each h2 element on the page, but it assumes that the [`offsetParent`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/offsetParent) for each h2 is the page's `main` element. However, I put the headings inside each table, which makes the table the h2's offset parent, and breaks the position comparison logic.

The quickest way to address this would be to pull the headings out of the table captions and make them match the nav that we have on other pages. That makes the web experience more consistent with the rest of the site but comes at the expense of visual consistency with the PDF version of the schedule. I'm attaching a screenshot of what that looks like in my local env.

Screenshot 2024-08-27 162714

If that's not an acceptable trade-off, I can work on updating the setCurrentLink logic to behave differently when the offsetParent is a table. The downside is that it's a hack and it risks introducing bugs into the sub-nav behavior for the existing guides (i.e. it will take longer to manually test).

File: content/derisking-government-tech/resources/kickoff.md (hunk 1 above)
Comment (nateborr) [reply content/derisking-government-tech/resources/kickoff.md]: I've updated these to `h2`s.

File: content/derisking-government-tech/resources/open-source-security.md (hunk 3 above)
Comment (nateborr) [reply content/derisking-government-tech/resources/open-source-security.md]: Sorry I missed the bugs in the side-nav behavior. The issue on the "Market research" and "Verbal interview questions" pages is that I used `h3`s rather than `h2`s to match the subheading font style in the design. Semantically, I should use `h2`s.

I'll make that change; if needed I can update the h2 style for those specific pages, although I'd like to try it out without that step first.

Comment (nateborr) [reply content/derisking-government-tech/resources/open-source-security.md]: I pushed up the h3-to-h2 change for those pages and updated the PR description accordingly.

File: content/derisking-government-tech/resources/kickoff.md (hunk 2 above)
Comment (melchoyce) [reply content/derisking-government-tech/resources/kickoff.md:24]: I think the solution in the screenshot is fine, thanks for exploring!

**Review by melchoyce — APPROVED** [review]

File: content/derisking-government-tech/resources/quality-indicators.md [hunk 5]
```diff
@@ -1,9 +1,71 @@
...
+    </tr>
+    <tr>
+      <td>Properly styled code</td>
+      <td>Meets acceptable quality level</td>
+      <td>0 linting errors and 0 warnings</td>
+      <td>Styling standards and linters</td>
+    </tr>
+    <tr>
+      <td>Accessibility</td>
+      <td><a href={{ "https://www.w3.org/TR/WCAG21/" | url }}>Web Content Accessibility Guidelines 2.2 – ‘AA’ standards</a></td>
```
Comment (peterrowland) [inline content/derisking-government-tech/resources/quality-indicators.md:42]: The link is still pointing to WCAG 2.1, but should point to https://www.w3.org/TR/WCAG22/

**Review by peterrowland — APPROVED** [review]
looks good and everything works as expected. Suggested one small change to update link to WCAG 22 from WCAG 21

File: content/derisking-government-tech/resources/quality-indicators.md (hunk 5 above)
Comment (nateborr) [reply content/derisking-government-tech/resources/quality-indicators.md:42]: Good catch, I'll update that.
